test/gradient.jl

```diff
@@ -206,12 +206,26 @@ using Test, Interpolations, DualNumbers, LinearAlgebra
         dims = [1, 2, 3]
         for dt in dtypes
             for dm in dims
-                itp = interpolate(rand(dt, ntuple(_ -> 5, dm)...), BSpline(Linear()))
-                etp = extrapolate(itp, rand(dt))
-                @test all(iszero, @inferred(Interpolations.gradient(etp, 6., 6.)))
-                @test all(iszero, @inferred(Interpolations.gradient(etp, -6., 6.)))
-                @test all(iszero, @inferred(Interpolations.gradient(etp, 6., -6.)))
-                @test all(iszero, @inferred(Interpolations.gradient(etp, -6., -6.)))
+                itp_n = interpolate(rand(dt, ntuple(_ -> 5, dm)...), BSpline(Linear()))
+                etp_n = extrapolate(itp_n, rand(dt))
+                if dm==2
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, 6., 6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, -6., 6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, 6., -6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, -6., -6.)))
+                elseif dm==1
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, 6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, -6.)))
+                else
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, 6., 6., 6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, -6., 6., 6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, 6., -6., 6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, -6., -6., 6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, 6., 6., -6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, -6., 6., -6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, 6., -6., -6.)))
+                    @test all(iszero, @inferred(Interpolations.gradient(etp_n, -6., -6., -6.)))
+                end
```

Comment:
These two seem sufficient to me.

```julia
for x in [ntuple(_->0, dm), ntuple(_->6, dm)]
```

Reply:
But this will fall short for the 2-dim and 3-dim cases where we have multiple quadrants right? It's better to get it checked for every quadrant right? That is why I hardcoded them.

Reply:
Yeah I get your points but it's not really necessary. We only need to choose _some_ out of range points and test it. This is a more programmable way and is more friendly for future maintainers.

Reply:
Ok I have changed it.